Client/Frontend/Browser/TabManager.swift

```diff
@@ -417,8 +417,8 @@ class TabManager: NSObject {
             if tabIndex == viableTabs.count {
                 tabIndex -= 1
             }
-            if tabIndex < viableTabs.count && !viableTabs.isEmpty {
-                _selectedIndex = tabs.index(of: viableTabs[tabIndex]) ?? -1
+            if let currentTab = viableTabs[safe: tabIndex], tabIndex < viableTabs.count && !viableTabs.isEmpty {
```

Comment:
If the `safe index` extension checks for bounds, do you need `tabIndex < viableTabs.count && !viableTabs.isEmpty` anymore?